kafka_consumer/datadog_checks/kafka_consumer/client/confluent_kafka_client.py

```diff
@@ -12,13 +12,41 @@ class ConfluentKafkaClient(KafkaClient):
     @property
     def kafka_client(self):
         if self._kafka_client is None:
-            self._kafka_client = AdminClient(
-                {
-                    "bootstrap.servers": self.config._kafka_connect_str,
-                    "socket.timeout.ms": self.config._request_timeout_ms,
-                    "client.id": "dd-agent",
+            config = {
+                "bootstrap.servers": self.config._kafka_connect_str,
+                "socket.timeout.ms": self.config._request_timeout_ms,
+                "client.id": "dd-agent",
+                "security.protocol": self.config._security_protocol,
+            }
+
+            if self.config._sasl_mechanism == "OAUTHBEARER":
+                oauth_config = {
+                    "sasl.mechanism": self.config._sasl_mechanism,
+                    "sasl.oauthbearer.method": "oidc",  # required to be `oidc` for OAUTHBEARER token login
+                    "sasl.oauthbearer.client.id": self.config._sasl_oauth_token_provider.get("client_id"),
+                    "sasl.oauthbearer.token.endpoint.url": self.config._sasl_oauth_token_provider.get("url"),
+                    "sasl.oauthbearer.client.secret": self.config._sasl_oauth_token_provider.get("client_secret"),
                 }
-            )
+                config.update(oauth_config)
+
+            if self.config._sasl_mechanism == "GSSAPI":
+                kerb_config = {
+                    "sasl.mechanism": self.config._sasl_mechanism,
+                    "sasl.username": self.config._sasl_plain_username,
+                    "sasl.password": self.config._sasl_plain_password,
+                    "sasl.kerberos.service.name": self.config._sasl_kerberos_service_name,
+                    # TODO: _sasl_kerberos_domain_name doesn't seem to be used?
+                    # looks like in kafka-python, sasl_kerberos_domain name can also be the host address:
+                    # https://github.com/dpkp/kafka-python/blob/7ac6c6e29099ccba4d50f5b842972dd7332d0e58/kafka/conn.py#L712
+                    # TODO: If `sasl.kerberos.keytab` is not configured, does it automatically
```

Comment:
TODO: Discuss

Reply:
Attempt to read `KRB5_CLIENT_KTNAME` env var, if exists, then send to confluent-kafka and leave a warning that this feature will be deprecated eventually. We can also leave a section in the troubleshooting section to help support.